src/Credentials.js

```diff
@@ -1,20 +1,59 @@
 import { createJWT, verifyJWT } from './JWT'
 import UportLite from 'uport-lite'
 import nets from 'nets'
+import { isMNID, encode, decode } from 'mnid'
+
+const networks = {
+  'mainnet':   {  id: '0x1',
+                  registry: '0xab5c8051b9a1df1aab0149f8b0630848b7ecabf6',
+                  rpcUrl: 'https://mainnet.infura.io' },
+  'ropsten':   {  id: '0x3',
+                  registry: '0x41566e3a081f5032bdcad470adb797635ddfe1f0',
+                  rpcUrl: 'https://ropsten.infura.io' },
+  'kovan':     {  id: '0x2a',
+                  registry: '0x5f8e9351dc2d238fb878b6ae43aa740d62fc9758',
+                  rpcUrl: 'https://kovan.infura.io' }
+  // 'infuranet': {  id: '0x2a'
+  //                 registry: '',
+  //                 rpcUrl: 'https://infuranet.infura.io' }
+}
+
+const DEFAULTNETWORK = 'kovan'
+
+const configNetwork = (net = DEFAULTNETWORK) => {
+  if (typeof net === 'object') {
+    ['id', 'registry', 'rpcUrl'].forEach((key) => {
+      if (!net.hasOwnProperty(key)) throw new Error(`Malformed network config object, object must have '${key}' key specified.`)
+    })
+    return net
+  } else if (typeof net === 'string') {
+    if (!networks[net]) throw new Error(`Network configuration not available for '${net}'`)
+    return networks[net]
+  }
+
+  throw new Error(`Network configuration object or network string required`)
+}
 
 export default class Credentials {
   constructor (settings = {}) {
-    this.settings = settings
-    if (!this.settings.registry) {
-      const registry = UportLite()
-      this.settings.registry = (address) => new Promise((resolve, reject) => {
-        registry(address, (error, profile) => {
-          if (error) return reject(error)
-          resolve(profile)
-        })
+  this.settings = settings
+  this.settings.network = configNetwork(settings.network)
+  this.net = this.settings.network
+  if (settings.address) {
+    this.settings.address = isMNID(settings.address) ? settings.address : encode({network: this.net.id, address: settings.address})
+  }
+  if (!this.settings.registry) {
+    const registryNetwork = {[this.net.id]: {registry: this.net.registry, rpcUrl: this.net.rpcUrl}}
+    const registry = UportLite({networks: registryNetwork})
+    this.settings.registry = (address) => new Promise((resolve, reject) => {
+      const addressMNID = isMNID(address) ? address : encode({network: this.net.id, address})
```

Comment:
Actually here just pass the raw address into uport-lite. It uses it to handle fallback to the old registry on ropsten.